Python/bltinmodule.c

```diff
@@ -1928,10 +1928,12 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
 
         stdin_encoding = _PyObject_GetAttrId(fin, &PyId_encoding);
         stdin_errors = _PyObject_GetAttrId(fin, &PyId_errors);
-        if (!stdin_encoding || !stdin_errors)
+        if (!stdin_encoding || !stdin_errors ||
+            !PyUnicode_Check(stdin_encoding) ||
+            !PyUnicode_Check(stdin_errors))
```

Comment:
Isn’t this already covered by the PyUnicode_AsUTF8 (aka _PyUnicode_AsString in 3.5) error checking below?

Also, I would use a double indent here to distinguish the “if” condition from the body.